src/app/home/pages/PlansPage/components/Wizard/WizardFormik.tsx

```diff
@@ -20,129 +23,136 @@ const WizardFormik: React.FunctionComponent<IWizardFormikProps> = ({
-            targetPVCName = editedPV ? editedPV.newPVCName : mappedPVCNameArr[1];
+            return {
+              sourcePVCName,
+              targetPVCName,
+              pvName,
+              pvcNamespace,
+            };
           }
-          return {
-            sourcePVCName,
-            targetPVCName,
-            pvName,
-            pvcNamespace,
-          };
-        } else {
-          return {
-            sourcePVCName,
-            targetPVCName,
-            pvName,
-            pvcNamespace,
-          };
-        }
-      });
-      const hasDuplicatePVMapping = existingPVCNameMap.find((pv, index) => {
-        const editedPVCName = values?.currentTargetPVCName?.name;
-        const editedPVCNameAssociatedPVName = values?.currentTargetPVCName?.srcPVName;
-        const isIntraClusterPlan = values.sourceCluster === values.targetCluster;
+        });
+        const hasDuplicatePVMapping = existingPVCNameMap.find((pv, index) => {
+          const editedPVCName = values?.currentTargetPVCName?.name;
+          const editedPVCNameAssociatedPVName = values?.currentTargetPVCName?.srcPVName;
+          const isIntraClusterPlan = values.sourceCluster === values.targetCluster;
 
-        return (
-          (editedPVCName === pv?.targetPVCName && editedPVCNameAssociatedPVName !== pv.pvName) ||
-          (editedPVCName === pv?.sourcePVCName && editedPVCNameAssociatedPVName !== pv.pvName) ||
-          (editedPVCName === pv?.sourcePVCName && isIntraClusterPlan)
-        );
-      });
+          return (
+            (editedPVCName === pv?.targetPVCName && editedPVCNameAssociatedPVName !== pv.pvName) ||
+            (editedPVCName === pv?.sourcePVCName && editedPVCNameAssociatedPVName !== pv.pvName) ||
+            (editedPVCName === pv?.sourcePVCName && isIntraClusterPlan)
+          );
+        });
 
-      const targetPVCNameError = utils.testTargetName(values?.currentTargetPVCName?.name);
-      if (targetPVCNameError !== '') {
-        errors.currentTargetPVCName = targetPVCNameError;
-      } else if (hasDuplicatePVMapping) {
-        errors.currentTargetPVCName =
-          'A mapped target pvc with that name already exists. Enter a unique name for this target pvc.';
-      }
+        const targetPVCNameError = utils.testTargetName(values?.currentTargetPVCName?.name);
+        if (targetPVCNameError !== '') {
+          errors.currentTargetPVCName = targetPVCNameError;
+        } else if (hasDuplicatePVMapping) {
+          errors.currentTargetPVCName =
+            'A mapped target pvc with that name already exists. Enter a unique name for this target pvc.';
+        }
 
-      return errors;
-    }}
-    onSubmit={() => {
-      return null;
-    }}
-    validateOnBlur={false}
-    enableReinitialize
-  >
-    {children}
-  </Formik>
-);
+        return errors;
+      }}
+      onSubmit={(values) => {
+        dispatch(PlanActions.updateCurrentPlanStatus({ state: CurrentPlanState.Pending }));
```

Comment:
Move the submit logic here so that values can be set in the wizard onMove function & reliably read from formik here in their final state.